File: src/main/java/com/blockout/clubs/models/entities/Club.java
package com.blockout.clubs.models.entities;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "clubs")
public class Club {

    @Id
    @Column(name = "id", nullable = false, unique = true)
    private String id;

    @Column(name = "raw_name", nullable = false)
    private String rawName;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "city")
    private String city;

    @Column(name = "postal_code")
    private String postalCode;

    @Column(name = "email")
    private String email;

    @Column(name = "phone_number")
    private String phoneNumber;

    @Column(name = "website")
    private String website;

    @Column(name = "logo_url")
    private String logoUrl;

    @Builder.Default
    @Column(name = "active", nullable = false)
    private Boolean active = true;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "last_update")
    private LocalDateTime lastUpdate;

    @PrePersist
    public void prePersist() {
        createdAt = LocalDateTime.now();
        lastUpdate = LocalDateTime.now();
    }

    @PreUpdate
    public void preUpdate() {
        lastUpdate = LocalDateTime.now();
    }
}

File: apps/frontend/mobile/src/modules/club/model/Club.ts
export type ClubResponse = {
  id: string;
  rawName: string;
  name: string;
  address: string | null;
  city: string | null;
  postalCode: string | null;
  email: string | null;
  phoneNumber: string | null;
  website: string | null;
  logoUrl: string | null;
  active: boolean;
  latitude: number | null;
  longitude: number | null;
  createdAt: string | null;
  lastUpdate: string | null;
};

export type UpdateClubRequest = {
  rawName?: string;
  name?: string;
  address?: string | null;
  city?: string | null;
  postalCode?: string | null;
  logoUrl?: string | null;
  email?: string | null;
  phoneNumber?: string | null;
  website?: string | null;
};
